lightningd/chaintopology.c

```diff
@@ -86,10 +88,22 @@ static void filter_block_txs(struct chain_topology *topo, struct block *b)
 		owned = AMOUNT_SAT(0);
 		txid = b->txids[i];
 		if (txfilter_match(topo->bitcoind->ld->owned_txfilter, tx)) {
+			// fprintf(stderr, "CAG txfilter_match found \n");
+			fprintf(stderr, "CAG txfilter_match found %s\n", fmt_bitcoin_txid(tmpctx, &txid));
+			size_t k;
 			wallet_extract_owned_outputs(topo->bitcoind->ld->wallet,
 						     tx->wtx, is_coinbase, &b->height, &owned);
 			wallet_transaction_add(topo->ld->wallet, tx->wtx,
 					       b->height, i);
+			// invoice_check_onchain_payment(tx);
+			for (k = 0; k < tx->wtx->num_outputs; k++) {
+			const u8 *oscript = bitcoin_tx_output_get_script(tmpctx, tx, k);
+				if (txfilter_scriptpubkey_matches(topo->bitcoind->ld->owned_txfilter, oscript)) {
```

Comment:
Don't need to check this twice...

Reply:
What do you mean?

First we check to see if the tx matches. Then we have to find which outpoint inside the tx is ours (txfilter_scriptpubkey_matches is a new function that does this) so we can check to see if the amount is sufficient to satisfy the invoice (and mark which outpoint was used, for our records). Probably could be cleaned up a bit, but I'm not immediately seeing a place where we check anything twice. It's early though, so perhaps I missed something :)

Reply:
If you mean that we extract our owned outputs in `wallet_extract_owned_outputs`, yes, that is true. I needed to decide whether to alter `wallet_extract_owned_outputs` to return the owned outpoints or just scan the tx again, processing each owned outpoint as it's found, and the latter seemed cleaner. Lmk if you want it the other way.